### PR TITLE
feat: add linter + formater as github action

### PR DESCRIPTION
**Pull request by mathe42**
I use the same config (+ tests) for own project.

### CODE REVIEW

**Review by oberbeck — APPROVED** [review]

**Comment by VixieTSQ** [comment]
Wonder if it would be worth testing in bun, node, and deno because we support all of them and therefore should confirm it works for all of them.

**Comment by mathe42** [comment]
@VixieTSQ I don't agree as this is (currently) only linting and formating....

**Comment by mathe42** [comment]
@tobiemh ready for review (and merge?). I use the same config in my own project so it should be fine. It currently doesn't run as it is just in the fork repo.

File: deno.json [hunk 1]
```diff
@@ -0,0 +1,12 @@
+{
+  "lint": {
+    "files": {
+      "include": ["src/", "./mod.ts", "./rollup.config.ts"]
+    }
+  },
+  "fmt": {
```
Comment (tobiemh) [inline deno.json:7]: @mathe42 can we use tabs here? And then re-fmt the PR?

```js
"options": {
      "useTabs": true,
      "lineWidth": 80,
      "indentWidth": 4,
      "singleQuote": false,
 }
```

Comment (mathe42) [reply deno.json]: Done!

**Comment by VixieTSQ** [comment]
> @VixieTSQ I don't agree as this is (currently) only linting and formating....

Ah sorry. Thought this included tests.

**Pull request merged**